### DISCUSSION TITLE
Yolo5 output fps   getting lower  while  running  detect.py

### DISCUSSION THREAD

**Discussion by samehsy** [ultralytics/yolov5#7925] [General]
Hello ..
I was  working  on safety helmet  detection using  yolo5  and  I  create a method  to  convert   yolo5  output  to  FFmpeg stream  to  make it then  as HLS  stream  ... but  the  weird  thing is that  I get low fps   from  Yolo  detection also it's not  constant  fps  it's  still  decreasing  while  running  
my  main  question  is whether the training  may  cause  this behavior  and  if yes what  I  should do  to  get stable   fps 
in  the  above  photo the output of the  FFmpeg streaming
![Screenshot from 2022-05-22 23-07-14](https://user-images.githubusercontent.com/40625954/169714013-515deb6c-cbaf-4804-8b5f-ef1ec80b4aeb.png)

**Comment by glenn-jocher** [comment]
👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃

**Reply by samehsy** [reply]
it's really  the  same default code  in the detect.py  script  just  I  use method  to  output  result  in  each  frame  to  ffmpeg  stream  
bellow  I  will  put the  code for  FFmpeg 
here is   my detect.py  script 
[https://github.com/samehsy/YOLO5-FFMPEG.git](url)

I  call  this method  when  initialize  detect script inside the run method  
`
def run_ffmpeg(width, height, fps):
 ffmpg_cmd = [
        'ffmpeg',
        '-y',
        '-f', 'rawvideo',
        '-vcodec', 'rawvideo',
        '-pix_fmt', 'bgr24',
        '-s', "{}x{}".format(width, height),
        '-r', str(fps),
        '-i', '-', 
        '-hls_time', '5',
        '-hls_list_size', '6',
        '{streamResult}/hls/index.m3u8'
    ]
    return subprocess.Popen(ffmpg_cmd, stdin=subprocess.PIPE)
`
here is where   I  call the method 
` 
     # Dataloader
    if webcam:
        view_img = check_imshow()
        cudnn.benchmark = True  # set True to speed up constant image size inference
        dataset = LoadStreams(source, img_size=imgsz, stride=stride, auto=pt)
        bs =30   # batch_size

        ffmpeg_process = run_ffmpeg( 640,480, 30) # image size  and  frames
`
then  white model  running  I  pass  the result  to  stream as  bellow
 ` # Stream results
            im0 = annotator.result()
            fmpeg_process.stdin.write(im0 )
            #if view_img:
            #     cv2.imshow(str(p), im0)
            #     cv2.waitKey(1)  # 1 millisecond`
